sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -70,63 +65,52 @@ case class GpuAnd(left: Expression, right: Expression) extends CudfBinaryOperato
 
   override def binaryOp: BinaryOp = BinaryOp.LOGICAL_AND
 
-  import GpuLogicHelper._
-
-  // The CUDF implementation of `and` will return a null if either input is null
-  // Spark does not.
-  // |LHS/RHS| TRUE  | FALSE | NULL  |
-  // |TRUE   | TRUE  | FALSE | NULL  |
-  // |FALSE  | FALSE | FALSE | FALSE |
-  // |NULL   | NULL  | FALSE | NULL  |
-  // So we have to make some adjustments.
-  //  IF (A <=> FALSE, FALSE,
-  //    IF (B <=> FALSE, FALSE,
-  //      A cudf_and B))
-
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
     val l = lhs.getBase
     val r = rhs.getBase
-    withResource(Scalar.fromBool(false)) { falseVal =>
-      val firstPass = withResource(l.and(r)) { lAndR =>
-        withResource(l.equalToNullAware(falseVal)) { lIsFalse =>
-          lIsFalse.ifElse(falseVal, lAndR)
-        }
-      }
-      withResource(firstPass) { firstPass =>
-        withResource(r.equalToNullAware(falseVal)) { rIsFalse =>
-          rIsFalse.ifElse(falseVal, firstPass)
+    withResource(new Table(l, r)) {
+      table => {
+        val astOp = new ast.BinaryOperation(ast.BinaryOperator.NULL_LOGICAL_AND,
+          new ast.ColumnReference(0),
+          new ast.ColumnReference(1))
+        withResource(astOp.compile()) { compiledAstExpr =>
+          compiledAstExpr.computeColumn(table)
```

Comment:
I wrote identical code to this in https://github.com/NVIDIA/spark-rapids/commit/ee162b52d2c12702de5df9af53e1ac5e21563748 and I wonder if we should consider having some common utility code around building these AST expressions. Another option would be exposing this logic in object `GpuLogicHelper` so that it can more easily be re-used elsewhere.

Reply:
@revans2 does creating common utility code for this fall under your larger task to use AST for more operations?  I can certainly move this to a utility function that accepts the op as an argument.  Also - my current patch removes `GpuLogicHelper`, because I removed the use of `egNullAware()` by GpuAnd and GpuOr.  Do we need to keep it around in case others need it?